### PR TITLE
ERC-2477

### PR DESCRIPTION
**Pull request by MoMannn**
@fulldecent Requesting solidity code review.

### CODE REVIEW

**Comment by MoMannn** [comment]
> The number should be ERC-2477.
> 
> There is lots of legacy usage of "Imprint".
> 
> It seems like the design is changed so that individual tokens can be stored with different schemas (whereas before it was one schema for the whole ledger). Please confirm this is intentional.

@fulldecent The design for the xcert stays the same. So following single responsibility principle there is one schema per ledger but since schema is now on the contract following 2477 this means you can query it per token. This is now also reflected in the framework. Another thing naming wise is that on the contract we use naming appropriate to the 2477 standard while on the framework we kept the naming from before since we believe it is simpler to understand and it keeps consistency with previous releases.

File: packages/0xcert-ethereum-xcert-contracts/src/contracts/utils/selector.sol [hunk 1]
```diff
@@ -64,7 +59,7 @@ contract Selector
     returns (bytes4)
   {
     XcertMutable i;
```
Comment (fulldecent) [inline packages/0xcert-ethereum-xcert-contracts/src/contracts/utils/selector.sol]: Selectors can now directly be like:

```solidity
XcertMutable.updateTokenImprint.selector;
```

Comment (MoMannn) [reply packages/0xcert-ethereum-xcert-contracts/src/contracts/utils/selector.sol]: With 0.6.1 version this does not work.

**Comment by fulldecent** [comment]
Got it. Good to ship as is.

**Comment by xpepermint** [comment]
Ref: https://github.com/ethereum/EIPs/pull/2477

**Pull request merged**